### DISCUSSION TITLE
GPU implementation for cell2location.run_cell2location()?

### DISCUSSION THREAD

**Discussion by carloelle** [BayraktarLab/cell2location#54] [Q&A] [answered]
Dear Team,

thanks for the wonderful package!
I ran through your tutorial and API and i noticed that cell2location.run_regression() can work on CUDA (through the 'use_cuda': True flag), making the computations much faster, instead cell2location.run_cell2location() cannot. Is there any particular reason for that? If not would you be able to implement a GPU implementation for run_cell2location()?

Best,
Carlo

**Comment by vitkl** [comment, marked as answer]
Hi @carloelle 

Apologies for the slow reply. The two functions both support GPU but use different packages. The regression model is written in pure PyTorch and cell2location model is written using pymc3 (theano/aesera backend for GPU). This means that to use the GPU with `cell2location.run_cell2location` you need to do different things. Specifically note this message at the beginning of the notebook:
<img width="963" alt="Screenshot 2021-07-21 at 17 10 21" src="https://user-images.githubusercontent.com/22567383/126522611-013f5f9e-72a0-429e-ae8d-febfba3e93e6.png">

**Reply by carloelle** [reply]
Thanks!

**Reply by vitkl** [reply]
@carloelle Just as heads up that new version is up (from late August)

**Comment by vitkl** [comment]
We are changing this soon (weeks) so both parts of the workflow will be implemented with https://pyro.ai/ backend and https://scvi-tools.org/ interface.
